procrastinate/worker.py

```diff
@@ -131,18 +139,20 @@ def _log_job_outcome(
         else:
             log_action, log_title = "job_error", "Error"
 
-        duration = (
-            job_context.job_result.end_timestamp
-            - job_context.job_result.start_timestamp
-        )
-        text = (
-            f"Job {job_context.job.call_string} ended with status: {log_title}, "
-            f"lasted {duration:.3f} s"
-        )
+        text = f"Job {job_context.job.call_string} ended with status: {log_title}, "
+        if (
+            job_context.job_result.start_timestamp
+            and job_context.job_result.end_timestamp
+        ):
```

Comment:
This is an interesting one.
I am going to add a comment.

I don't think the `job_result` makes sense to be part of the `JobContext`. 
It will never be set for the task implementation to inspect, and it appears it is used only internally within procrastinate for logging purposes.
That said, I am not planning to remove it as part of this PR.